x/msgfees/keeper/keeper.go

```diff
@@ -71,6 +73,15 @@ func (k Keeper) GetFloorGasPrice(ctx sdk.Context) sdk.Coin {
 	return min
 }
 
+// GetUsdConversionRate returns the current usd to hash (1000000000nhash) conversion rate in cents
+func (k Keeper) GetUsdConversionRate(ctx sdk.Context) uint64 {
```

Comment:
Should this be a float? You can simplify conversion using powers of 10. This would also remove the branch statement from lines 189-195 by just using multiplication.

Reply:
sticking with integer based math is important here ... we don't want to use decimals since there can be determinism and accuracy issues.